src/inference/gibbs.jl

```diff
@@ -62,11 +78,116 @@ function Sampler(alg::Gibbs, model::Model, s::Selector)
         @warn("[Gibbs] extra parameters specified by samplers don't exist in model: $(setdiff(space, get_pvars(model)))")
     end
 
-    info[:samplers] = samplers
+    # Create a state variable.
+    state = GibbsState(model, samplers)
+
+    # Create the sampler.
+    spl = Sampler(alg, info, s, state)
+
+    # Add Gibbs to gids for all variables.
+    for sym in keys(spl.state.vi.metadata)
+        vns = getfield(spl.state.vi.metadata, sym).vns
+        for vn in vns
+            Turing.RandomVariables.updategid_forced!(spl.state.vi, vn, spl)
```

Comment:
L91 updates the `gids` for `spl`, which is the Gibbs sampler. L95 attaches each subsampler to it's respective variables. 

I only did it this way because of the way Gibbs sampling works now -- the Gibbs sampler has it's own native `vi` that overwrites the subsampler's `vi`, so there needed to be a "master" `VarInfo` that knows about all the samplers.

Reply:
But both are updating `spl.state.vi`. Should L95 update `local_spl.state.vi` or something?